pkg/ingester/user_state.go

```diff
@@ -259,7 +259,11 @@ func (u *userState) removeSeries(fp model.Fingerprint, metric labelPairs) {
 
 // forSeriesMatching passes all series matching the given matchers to the provided callback.
 // Deals with locking and the quirks of zero-length matcher values.
-func (u *userState) forSeriesMatching(ctx context.Context, allMatchers []*labels.Matcher, callback func(context.Context, model.Fingerprint, *memorySeries) error) error {
+// There are 2 callbacks. callback1 is called for each series, where the lock is held.
+// callback2 is called at certain intervals specified by callback2Interval,
+// i.e. callback2() after callback2Interval calls of callback1().
+func (u *userState) forSeriesMatching(ctx context.Context, allMatchers []*labels.Matcher, callback1 func(context.Context, model.Fingerprint, *memorySeries) error,
+	callback2 func(context.Context) error, callback2Interval int) error {
```

Comment:
Lets have better names for the callbacks - `append` and `send` would work for me.